### PR TITLE
[Enhancement]Refactor init default warehouse

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 1]
```diff
@@ -1157,6 +1160,10 @@ private void transferToLeader() {
                 initDefaultCluster();
             }
 
+            if (!isDefaultWarehouseCreated) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1163]: change `isDefaultWarehouseCreated` to true after initialized successful?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/LocalWarehouse.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|6|100.00%|[]|

**Review by wyb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/WarehouseManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/warehouse/LocalWarehouse.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|6|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**